README.md

```diff
@@ -1,2 +1,102 @@
-# scalajs-fs2-router
-Scalajs router for Single Page Webapplications based on fs2
+# fs2-spa-router
+
+Scalajs router for Single Page web-applications based on *fs2* using 
+*http4s-dsl*'s `Uri` and `Path` to calculate and represent routes.
+
+```scala
+"com.github.jberggg" %%% "fs2-spa-router" % "0.1.0-SNAPSHOT"
+```
+
+**Note**: CI/CD is not yet in place, you need to checkout the project and build 
+it yourself with `sbt publishLocal`.
+
+## How it works
+
+The router is built upon the [*History API*](https://developer.mozilla.org/en-US/docs/Web/API/History) 
+which allows to route on regular looking paths rather then a path section delimited
+by a `#`. 
+
+When navigating from within the SPA app, the path and state is pushed to the *History API*. 
+When the user presses the back button, the router will receive the previous path and state
+from the *History API* which then can be used  to reconstruct the view the user wants to navigate
+to. The library makes no assumption about the state pushed to and received from the *History Api*. 
+Thus it can be anything you want [as long as](https://developer.mozilla.org/en-US/docs/Web/API/History_API/Working_with_the_History_API):
+
+> The state object can be anything that can be serialized.
```

Comment:
This is fascinating :) I guess this won't work with ordinary Scala objects.

Reply:
Yes it won't work directly, you have to convert it to a `js.Object` first. I added the syntax `toJsObject` for that.

Reply:
I saw that. I might be confused but I'm not sure that will work. As far as I understand a Scala object cannot be serialized, so you cannot use it in that way. Something you could do is use e.g. Circe to encode a `case class` as a plain-old-javascript-object, and that would definitely be serializable.

Reply:
Maybe the js.Object can then be serialized... 

But to be honest, I never really used the state coming back 😂  Will give it a go tonight.

Reply:
You are right and it turned out that Circe has helper functions that lets you convert Scala Objects to / from Javascript with the help of a Codec. I'm going to provide some hints in the docs, but probably not as part of this PR :)